R/spark_version.R

```diff
@@ -18,8 +18,12 @@ spark_version_clean <- function(version) {
 spark_version <- function(sc) {
 
   # use cached value if available
-  if (!is.null(sc$spark_version))
-    return(sc$spark_version)
+  if (!is.null(sc$state$spark_version))
```

Comment:
This is nice! Get's called a bunch of times! I had to do this for Livy already (see https://github.com/rstudio/sparklyr/blob/3a3142d519026e294dc0f50d400cf052c5f8d1a6/R/livy_connection.R#L778),but it doesn't hurt to do this for shell connections as well! 👍